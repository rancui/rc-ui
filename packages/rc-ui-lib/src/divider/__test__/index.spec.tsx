import * as React from 'react';
import { cleanup, render } from '@testing-library/react';
import Divider from '..';

describe('Divider', () => {
  afterEach(() => {
    cleanup();
  });

  it('should render correctly', async () => {
    const { container } = render(<Divider />);
    expect(container).toMatchSnapshot();
  });

  it('should render correctly when have children', async () => {
    const { container } = render(<Divider>text</Divider>);
    expect(container.querySelector('.rc-divider--content-center').textContent).toEqual('text');
  });

  it('should render correctly when contentPosition is set for left', async () => {
    const { container } = render(<Divider contentPosition="left">text</Divider>);
    expect(container.querySelector('.rc-divider--content-left').textContent).toEqual('text');
  });

  it('should render correctly when contentPosition is set for right', async () => {
    const { container } = render(<Divider contentPosition="right">text</Divider>);
    expect(container.querySelector('.rc-divider--content-right').textContent).toEqual('text');
  });

  it('should render correctly when using dashed prop', async () => {
    const { container } = render(<Divider dashed>text</Divider>);
    expect(container.querySelector('.rc-divider--dashed').textContent).toEqual('text');
  });

  it('should render correctly when hairline is false ', async () => {
    const { container, rerender } = render(<Divider>text</Divider>);
    expect(container.querySelector('.rc-divider--hairline')).toBeTruthy();
    rerender(<Divider hairline={false}>text</Divider>);
    expect(container.querySelector('.rc-divider--hairline')).toBeFalsy();
  });
});
