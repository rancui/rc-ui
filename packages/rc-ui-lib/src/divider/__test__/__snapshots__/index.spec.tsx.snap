// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider should render correctly 1`] = `
<div>
  <div
    class="rc-divider rc-divider--hairline"
    role="separator"
  />
</div>
`;
